.github/workflows/ci.yml

```diff
@@ -0,0 +1,185 @@
+            exit 1
+          fi
+
+  check:
+    needs: version
+    if: always() && (needs.version.result == 'success' || needs.version.result == 'skipped')
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: true
+      matrix:
+        node-version: [14.x, 16.x, 18.x]
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Setup Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v3
+        with:
+          node-version: ${{ matrix.node-version }}
+
+      - name: Install dependencies
+        run: |
+          npm ci
+
+      - name: Build
+        run: |
+          npm run build
+
+      - name: Check
+        run: |
+          npm run check
+
+      - name: Test
+        run: npm run test:ci
+        env:
+          CI: true
+
+      - name: Upload code coverage
+        uses: codecov/codecov-action@v3
+        if: matrix.node-version == '18.x'
+
+  release:
+    needs: check
+    if: needs.check.result == 'success' && startsWith(github.ref, 'refs/tags/v')
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v3
+        with:
+          registry-url: "https://registry.npmjs.org"
+
+      - name: Install dependencies
+        run: |
+          npm ci
+
+      - name: Build Changelog
+        id: changelog
+        uses: mikepenz/release-changelog-builder-action@v2
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          configuration: "./.github/configs/changelog.json"
+          owner: MichalLytek
+          repo: type-graphql
+          failOnError: true
+
+      - name: Release
+        uses: softprops/action-gh-release@v1
+        with:
+          body: ${{ steps.changelog.outputs.changelog }}
+
+      - name: Publish
+        run: npm publish
+        env:
+          NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
+
+  website:
+    needs: check
+    if: needs.check.result == 'success' && github.event_name == 'push' && (github.ref == 'refs/heads/master' || startsWith(github.ref, 'refs/tags/v'))
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v3
+
+      - name: Install dependencies
+        working-directory: ./website
+        run: |
+          npm ci
+
+      - name: Build
+        working-directory: ./website
+        run: |
+          npm run build
+
+      - uses: peaceiris/actions-gh-pages@v3
+        with:
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          publish_dir: ./website/build/type-graphql
+          user_name: "github-actions[bot]"
+          user_email: "github-actions[bot]@users.noreply.github.com"
+          commit_message: |
+            Deploy website
```

Comment:
Why removed the `Deploy website version based on`?
It makes git message pretty printed:
![image](https://user-images.githubusercontent.com/10618781/219617982-c9523ebc-f674-46ea-a6f6-82407052aa92.png)

Reply:
Readd it maybe I deleted it while refactoring.
The one thing that is missing from the website job is to deploy it only when the docs or website directories are modified. This eliminates the need to deploy and build the website always when a commit is made

Reply:
I remember now.
It's because it was not obvious: "... Based on"??
I think it is way better to update it with:
"Based on ${{ github.sha }}"
It's clearer and we know that is based on what commit sha

Reply:
The `peaceiris/actions-gh-pages` is adding sha of `master` branch commit automatically, so that was my trick for pretty print it.

Reply:
Ok perfect. I didn't know that :)

Reply:
Fixed by `full_commit_message`